src/App.tsx

```diff
@@ -15,49 +17,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((previousGood, currentGood) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return previousGood.localeCompare(currentGood);
+
+      case SortType.LENGTH:
+        return previousGood.length - currentGood.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
```

Comment:
In React, when using the `useState` hook to manage state, it’s a common convention to name the state variable and its corresponding setter function using a specific naming template: `[variableName, “set” + variableName]`.

For example, in the following code:

```js
const [isReversed, setReversed] = useState(false);
```

`isReversed` is the state variable, and setReverse is the setter function for that variable.

Using this naming convention has several advantages:

- **Readability**: It makes it clear that the setReverse function is specifically for setting the `isReversed` state variable, which can improve the readability of the code.

- **Consistency**: Using a consistent naming convention makes it easier for other developers to understand the codebase and reduces the cognitive load of working with it.

- **Autocompletion**: Many code editors and IDEs will suggest the correct setter function name based on the state variable name, which can make it faster to write code.

Here’s an example of how we can use this naming convention with multiple state variables:

```js
const [isReversed, setIsReversed] = useState(false);
const [selectedItem, setSelectedItem] = useState(null);
const [count, setCount] = useState(0);
```

In summary, naming your React state variables and setter functions using the `[variableName, “set” + variableName]` template can improve the readability, consistency, and autocompletion of your code.